File: 1.js
function solution(numbers, hand) {
    let answer = '';
    
    let left = 10
    let right = 12
    
    for(let num of numbers) {
        
        if(num == 0)
            num = 11
        
        let column = num % 3
        
        if(column == 1) {
            answer += 'L'
            left = num
        }
        else if(column == 0) {
            answer += 'R'
            right = num
        }
        else if(column == 2) {
            const distance_L = distance(left, num)
            const distance_R = distance(right, num)
            
            if(distance_L < distance_R) {
                answer += 'L'
                left = num
            }
            else if(distance_L > distance_R) {
                answer += 'R'
                right = num
            }
            else if (distance_L == distance_R) {
                if(hand == "left") {
                    answer += 'L'
                    left = num
                }
                else if (hand == "right") {
                    answer += 'R'
                    right = num
                }
            }
        }
            
    }
    return answer;
}

function distance(n1, n2) {
    return Math.abs((n1 - 1) % 3 - (n2 - 1) % 3) + Math.abs(parseInt((n1 - 1) / 3) - parseInt((n2 - 1) / 3))
}

File: programers/정수 삼각형/java/1.java
class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        
        int[][] sum = new int[triangle.length][];
        for(int i = 0; i < triangle.length; i++)
            sum[i] = new int[i + 1];
        
        for(int i = 0; i < sum.length; i++) {
            for(int j = 0; j < sum[i].length; j++) {
                int left = 0;
                int right = 0;
                
                if(j != 0)
                    left = sum[i - 1][j - 1];
                if(i != j)
                    right += sum[i - 1][j];
                
                sum[i][j] = triangle[i][j];
                if(left > right)
                    sum[i][j] += left;
                else
                    sum[i][j] += right;
            }
        }
        
        for(int i = 0; i < sum.length; i++) {
            if(answer < sum[sum.length - 1][i])
                answer = sum[sum.length - 1][i];
        }
        
        return answer;
    }
}